### PR TITLE
Add more cancellation tests

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: tests/System.IO.Pipelines.Tests/ReadAsyncCancellationTests.cs [hunk 1]
```diff
@@ -338,6 +338,22 @@ public void ReadAsyncCompletedAfterPreCancellation()
...
+        public void ReadAsyncReturnsIsCancelOnCancelPendingReadBeforeGetResult()
+        {
+            var awaitable = Pipe.Reader.ReadAsync();
+            Assert.False(awaitable.IsCompleted);
+            awaitable.OnCompleted(() => {});
+
+            Pipe.Writer.WriteAsync(new byte[] {});
+            Pipe.Reader.CancelPendingRead();
+
+            Assert.True(awaitable.IsCompleted);
```
Comment (halter73) [inline tests/System.IO.Pipelines.Tests/ReadAsyncCancellationTests.cs]: Why isn't this assert in the other test?

Comment (pakrym) [inline tests/System.IO.Pipelines.Tests/ReadAsyncCancellationTests.cs]: No reason

File: tests/System.IO.Pipelines.Tests/FlushAsyncCancellationTests.cs [hunk 2]
```diff
@@ -215,6 +215,25 @@ public void FlushAsyncCompletedAfterPreCancellation()
...
+        {
+            var writableBuffer = Pipe.Writer.Alloc(MaximumSizeHigh);
+            writableBuffer.Advance(MaximumSizeHigh);
+            var awaitable = writableBuffer.FlushAsync();
+
+            Assert.False(awaitable.IsCompleted);
+            awaitable.OnCompleted(() => {});
+
+            Pipe.Reader.Advance(Pipe.Reader.ReadAsync().GetResult().Buffer.End);
+            Pipe.Writer.CancelPendingFlush();
```
Comment (halter73) [inline tests/System.IO.Pipelines.Tests/FlushAsyncCancellationTests.cs]: I think it makes sense to test flipping the order of the above two lines to verify Advance won't clobber the canceled state.

Comment (pakrym) [inline tests/System.IO.Pipelines.Tests/FlushAsyncCancellationTests.cs]: Adding

File: tests/System.IO.Pipelines.Tests/FlushAsyncCancellationTests.cs [hunk 3]
```diff
@@ -215,6 +215,44 @@ public void FlushAsyncCompletedAfterPreCancellation()
         }
 
         [Fact]
+        public void FlushAsyncReturnsIsCancelOnCancelPendingFlushBeforeGetResult()
```
Comment (cesarbs) [inline tests/System.IO.Pipelines.Tests/FlushAsyncCancellationTests.cs]: nit: `ReturnsIsCancelled`

**Pull request merged**